<h2>7.4连接到Mysql</h2>

<h3>问题</h3>
当从MySQL 入数据时,Sqoop引发了一个关于通信故障的异常。

```
ERROR manager.SqlManager: Error executing statement:
com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: Communications link
failure
```

<h3>解决方案</h3>
首先,排除用户通过网络访问数据库的连接性和权限问题。可能需要在JDBC设置属性interactiveClient=true,连接字符串或增加MySQL上的wait_timeout属性的值
在服务器端。

<h3>讨论</h3>
验证是否可以从运行的节点连接到数据库Sqoop通过在shell中使用以下命令:

```
mysql --host=<IP Address> --database=test --user=<username> --password=<password>
```

如果此工作正常,则它排除了客户端网络配置或安全/身份验证配置的任何问题。请注意,Sqoop还将需要来自Hadoop群集中所有节点的数据库连接。
在MySQL配置选项wait_timeout在空闲时间过长时可能导致连接关闭。当Sqoop在客户端重用相同的连接时,如果wait_timeout属性的值过低, 则可能会
遇到通信失败。一种解决方案是在JDBC连接字符串中设置属性interactiveClient=true, 使用备用超时周期。另一种解决办法是增加MySQL端wait_timeout
属性的值。